src/btree/bt_discard.c

```diff
@@ -245,20 +245,31 @@ __free_page_modify(WT_SESSION_IMPL *session, WT_PAGE *page)
 void
 __wt_ref_addr_free(WT_SESSION_IMPL *session, WT_REF *ref)
 {
+    WT_PAGE *home;
     void *ref_addr;
 
+    home = ref->home;
```

Comment:
I'm confident this doesn't need to be an ordered read, but (a) are the other readers doing an ordered read of `ref_addr`, (b) are the places that write this doing an ordered write?

Reply:
a) The answer to this question isn't an easy one, certain subsystems such as compaction expect to have exclusive access I believe and then don't need to use the barrier primitives. Additionally in places where we have allocated a new ref we don't write or read with barriers. In the split path where it is assumed that other things can be happening in the tree there are ordered reads and compare-and-swaps (which would be in place of the ordered write). But I can't say for sure if they are in every place of the code. I think that is best left for the shared variable review project.
b) I don't think we write `ref->home `with an ordered write in any places in the code. Even the `split_ref_prepare` path which is implicated in this bug doesn't. Again probably a concern for the shared variable review project.

Reply:
Ack, thanks.